### ISSUE TITLE
Synology NAS issues

### ISSUE THREAD

**Issue by wheeleran** [tomsquest/docker-radicale#165]
hi all. I'm having all sort of issues getting this running properly on Container Manager. I'm happy for BUILD_UID and GID to remain the default 2999, but mapping the /config folder doesn't take ownership of the folder and the container immediately stops. it only runs if i remove the /config folder mapping, which then means that any new login of any user with any password creates a new collection, and that's what i'm trying to lock down and restrict with a config, users, and rights file. i've tried changing UID and PID to a syn admin user, and setting the TAKE_FILE_OWNERSHIP to false, but same issue; any attempt to extend the very basic unsecure image crashes the container. does anyone have a synology-specific ELI5 and docker-compose that i can use - which uses a config, users, and rights file? thanks in advance!

**Comment by tomsquest** [comment]
Hi Wheelran,

Quite hard for me without having access to a Synology, even more to what they do there.

What you can absolutely try is to run the container as `root`, without any `user` option, and try without any limitation of perms. Somehow, start "open", then secure it a bit at a time.

**Comment by wheeleran** [comment]
I've tried that, but the problem is as soon as I try to add a /config folder - basic vanilla container runs fine. I tried your Option 1, but 2999 never takes ownership of the folder, and the BUILD_* ownership option never starts the container. Unfortunately the last Pull request in any way related to Synology is from 2019, and your README did mention Synology once, but I didn't understand the reference to PR#68. I'll try again, and thanks for replying!

**Comment by tomsquest** [comment]
Hi @wheeleran ,

Couple of things :
- what is the output of radicale (the log) when you have the problem?
- is the volume for the config, or the config directory with the config file created before hand (before "adding" it to the container?

About how it is working:
- Radicale needs a config in /config/config and a data directory
- The config should be readable, and the data writable
- The container does not alter anything in the config folder, but can will to put the data folder writable when running as root and `TAKE_FILE_OWNERSHIP` is true (the default)

**Comment by wheeleran** [comment]
i actually think i got it, but can i check one thing? is the expected error message a red "Error: 401" when trying to log in with a user not added to the 'users' file? if that IS correct, does the line 'realm = Radicale - Password Required' in config modify the text of that error message, or is that for something else?

**Comment by tomsquest** [comment]
> 'realm = Radicale - Password Required' in config modify the text of that error message, or is that for something else?
I don't know how if Radicale uses that config precisely for error message.

Enabling more logging would definetely help you see what does Radicale.

For my information, what was the issue you figured out?

**Comment by wheeleran** [comment]
i'm trying to pull some logs for you, but having a few issues. what i DO see when i run '--details' is multiple '[2024-11-23 08:25:17 +0000] [7/Thread-31866 (process_request_thread)] [INFO] GET response status for '/' in 0.000 seconds: 302 Found' for a bunch of different threads.

Yes i created a /config folder and a basic txt 'config' and 'users' file inside it. the config file is:

```[server]
hosts = 0.0.0.0:5232

[auth]
type = htpasswd
htpasswd_filename = /config/users
htpasswd_encryption = plain

[storage]
filesystem_folder = /data/collections
```

the users file is:

```username:password-plain-text
```

i tried a number of different versions of the env vars following [this](https://github.com/tomsquest/docker-radicale?tab=readme-ov-file#custom-usergroup-id-for-the-data-volume) section of the README. Option 3, with the suggestion that if my be a fix for Synology NAS using BUILD_UID etc did not work for me; the container would not stay running. Options 1 and 2 the container would either not start, or would start but not give me a login screen. Other tweaks i tried would run, but not read the users restriction in the 'users'  file.  the only thing that works, so far, is Option 0. the /data folder owner is 2999, the /config folder owner is me, the Synology admin user who created the folder. The docker-compose is your vanilla YAML [here](https://github.com/tomsquest/docker-radicale/blob/master/docker-compose.yml) with no deploy: or healthcheck:, and with the /config folder mapped.

this all works, no users except those in the 'users' file can log in or make a collection - which was exactly what i wanted - but i'm just wondering if the 'Error: 401' is the expected message when i do try to log in or create an unauthorised user, or if it means something else is wrong.

**Comment by tomsquest** [comment]
> i'm just wondering if the 'Error: 401' is the expected message when i do try to log in or create an unauthorised user, or if it means something else is wrong.

I do think it's the right error code for unauthorized access.
In Http, 401 is exactly that, so I would expect Radicale to return this type of error for a failed authorization.

So I think you're good :+1:
